dask/array/utils.py

```diff
@@ -427,12 +427,36 @@ def zeros_like_safe(a, shape, **kwargs):
     using the old behavior, returning ``np.zeros(shape, **kwargs)``.
     """
     try:
+        foo = np.zeros_like(a, shape=shape, **kwargs)
+        print("foo: ", type(foo))
         return np.zeros_like(a, shape=shape, **kwargs)
     except TypeError:
         kwargs.setdefault("dtype", _dtype_of(a))
         return np.zeros(shape, **kwargs)
 
 
+def _zeros_safe(shape, like):
+    """
+    Private zeros function intended to be used during graph construction
+    """
+    return zeros_safe(shape, like=like)
+
+
+def zeros_safe(*args, like, **kwargs):
+    """
+    Use the `like=` from `np.zeros` to create a new zeros dispatching
+    to the downstream library. If that fails, falls back to the
+    default NumPy behavior, resulting in a `numpy.ndarray`.
+    """
+    if like is None:
+        return np.zeros(*args, **kwargs)
+    else:
+        try:
+            return np.zeros(*args, like=meta_from_array(like), **kwargs)
+        except TypeError:
+            return np.zeros(*args, **kwargs)
+
+
```

Comment:
Why is this necessary? We have `zeros_like_safe` that handles the exact same case. IOW, these two are equivalent:

```
np.zeros_like(a, shape=(b, c))
np.zeros((b, c), like=a)
```

Reply:
You're right.  However, during graph construction I think we can only provide args (as opposed to kwargs). What do you think about keeping `_zeros_safe` and pointing it at `zeros_like_safe`.  This makes it possible to do the following:

> (_zeros_safe, a, (a.chunks[0][i], a.chunks[1][j]), meta_from_array(a))

Reply:
With `zeros_like_safe` the `shape` argument is the second positional argument, making it perfectly possible to just do:

```python
(zeros_like_safe, a, (a.chunks[0][i], a.chunks[1][j]))
```

or

```python
(zeros_like_safe, meta_from_array(a), (a.chunks[0][i], a.chunks[1][j]))
```

Not sure which of the above you actually need here.